src/context.c

```diff
@@ -140,6 +140,43 @@ static gchar* get_cmdline_bootname(void)
 	return bootname;
 }
 
+static gchar* get_custom_bootname(void)
+{
+	g_autoptr(GSubprocessLauncher) launcher = NULL;
+	g_autoptr(GSubprocess) handle = NULL;
+	g_autoptr(GDataInputStream) datainstream = NULL;
+	g_autoptr(GPtrArray) args_array = NULL;
+	g_autofree gchar *outline = NULL;
+	GInputStream *instream;
+
+	if (context->mock.custom_bootloader_output) {
+		return g_strdup(context->mock.custom_bootloader_output);
+	}
+
+	args_array = g_ptr_array_new();
+	g_ptr_array_add(args_array, context->config->custom_bootloader_backend);
+	g_ptr_array_add(args_array, (gchar *)("get-current"));
+
+	launcher = g_subprocess_launcher_new(G_SUBPROCESS_FLAGS_STDOUT_PIPE);
+	handle = r_subprocess_launcher_spawnv(launcher, args_array, NULL);
+
+	if (handle == NULL) {
+		return NULL;
+	}
+
+	instream = g_subprocess_get_stdout_pipe(handle);
+	datainstream = g_data_input_stream_new(instream);
+
+	outline = g_data_input_stream_read_line(datainstream, NULL, NULL, NULL);
+
+	g_subprocess_wait_check(handle, NULL, NULL);
+
+	if (outline)
+		return g_strdup(outline);
```

Comment:
When having checked `outline` before, better use `g_steal_pointer` here in case of success then.